src/MedTrackerAPI/Features/Devices/GetDevice.cs

```diff
@@ -0,0 +1,45 @@
+using MediatR;
+using MedTrackerAPI.Endpoints;
+using MedTrackerAPI.Infrastructure;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace MedTrackerAPI.Features.Devices;
+
+public static class GetDevice
+{
+    public record Response(int DeviceId, string Description, string Manufacturer, string Model, string PartNumber, string LotNumber);
+    
+    public sealed class Endpoint : IEndpoint
+    {
+        public void MapEndpoint(IEndpointRouteBuilder app)
+        {
+            app.MapGet("devices/{deviceId}", async (IMediator mediator, int deviceId) =>
+                {
+                    var query = new GetDeviceQuery { DeviceId = deviceId };
+                    var device = await mediator.Send(query);
+                    
+                    return device is null ? Results.BadRequest("Device not found") : Results.Ok(device);
+                }).WithTags("Devices");
+        }
+    }
+    
+    public class GetDeviceQuery : IRequest<Response>
+    {
+        public int DeviceId { get; set; }
+    }
+    
+    public class GetDeviceQueryHandler(MedTrackerDbContext context) : IRequestHandler<GetDeviceQuery, Response>
+    {
+        public async Task<Response> Handle(GetDeviceQuery request, CancellationToken cancellationToken)
```

Comment:
You can return `Response?` so that the compiler can give warnings to callers and you don't need the null forgiving (!) operator - aka damnit operator.